<h1>B09705039_劉惟恩_HW02

In [1]:
# 載入所需函示庫
import pandas as pd
import numpy as np
import scipy.stats as stats
import statsmodels.stats.api as sms
from matplotlib import pyplot as plt
%matplotlib inline
# 設定圖形大小; DPI越大圖越大
plt.rcParams["figure.dpi"] = 150

<h2>Problem 13.95

In [2]:
# 讀取資料集
df1 = pd.read_excel('Xr13-95.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，
# 並確認程式讀到的跟你在Excel中看到的一樣
print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


This Year  Last Year
0        189        151
1        225        224
2        106         30
3        146        132
4        234        233

Tail of dataset:


This Year  Last Year
35        104        102
36        275        242
37        137         82
38        128        148
39        159        179

<h3>a.

It implies that we should set the significance level to 0.1.

<h3>b.

令 This Year 的均數為 $\mu_1$。<br>
   Last Year 的均數為 $\mu_2$。

Null Hypothesis: $\mu_1 - \mu_2 <= 0$ <br>
Alternative Hypothesis: $\mu_1 - \mu_2 > 0$.

using t-test, paired samples

In [3]:
samp3 = df1['This Year'].values
samp4 = df1['Last Year'].values

samp_diff = samp3 - samp4
nobs = samp_diff.shape[0]
df = nobs - 1
print("degree of freedom = ", df)

tmpdesc = stats.describe(samp_diff)
t_value = tmpdesc.mean / (tmpdesc.variance ** 0.5) * (nobs ** 0.5)
print(f"t-value = {t_value:.4f}")
# tmpdesc

# p-values
ptmp = stats.t.cdf(t_value, df)
p_value_left = ptmp
p_value_right = 1 - ptmp
if ptmp > 0.5:
    ptmp = 1 - ptmp
p_value_twoside = ptmp * 2

print(f"p_value (two sided) = {p_value_twoside:.4f}")
print(f"p_value (right tail) = {p_value_right:.4f}")
print(f"p_value (left tail) = {p_value_left:.4f}")

degree of freedom =  39
t-value = 4.0775
p_value (two sided) = 0.0002
p_value (right tail) = 0.0001
p_value (left tail) = 0.9999


In our case, we are performing a right-tailed hypothesis test. The right-tailed p-value is 0.0001.
The p-value is smaller than the required significance level 0.1. As a result, we reject the null hypothesis in favor of the alternative hypothesis. 

We can conclude that advertising in the Yellow Pages improves sales.

In [4]:
t_alpha = stats.t.ppf(1 - 0.1 / 2, df)
std_xbar = (tmpdesc.variance / nobs) ** 0.5
LCL = tmpdesc.mean - t_alpha * std_xbar
UCL = tmpdesc.mean + t_alpha * std_xbar
print("Confidence interval = [%0.4f, %0.4f]" % (LCL, UCL))

Confidence interval = [11.5890, 27.9110]


<h3>c.

Satisfied

<h3>d.

No, it wouldn't be advantageous to perform this experiment with independent samples. Since we are not interested in the variation between annual sales among different stores, pairing observations by stores can eliminate the variation caused by different stores. By performing this experiment with independent samples will only increase the variation between annual sales among different stores, which won't do any good on this problem.

<h2>Problem 13.100

In [5]:
# 讀取資料集
df1 = pd.read_excel('Xr13-100.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，
# 並確認程式讀到的跟你在Excel中看到的一樣
print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


Company 1  Company 2
0      12859       9449
1       6082       5694
2       8604       7457
3       9230       7725
4      10405      10163

Tail of dataset:


Company 1  Company 2
50      11815       9631
51       6069       7798
52      12073      13252
53       6466       7274
54       9782      11055

令 Company 1 的均數為 $\mu_1$。<br>
   Company 2 的均數為 $\mu_2$。

Null Hypothesis: $\mu_1 - \mu_2 <= 0$ <br>
Alternative Hypothesis: $\mu_1 - \mu_2 > 0$.

using t-test, paired samples

In [6]:
samp3 = df1['Company 1'].values
samp4 = df1['Company 2'].values

samp_diff = samp3 - samp4
nobs = samp_diff.shape[0]
df = nobs - 1
print("degree of freedom = ", df)

tmpdesc = stats.describe(samp_diff)
t_value = tmpdesc.mean / (tmpdesc.variance ** 0.5) * (nobs ** 0.5)
print(f"t-value = {t_value:.4f}")
# tmpdesc

# p-values
ptmp = stats.t.cdf(t_value, df)
p_value_left = ptmp
p_value_right = 1 - ptmp
if ptmp > 0.5:
    ptmp = 1 - ptmp
p_value_twoside = ptmp * 2

print(f"p_value (two sided) = {p_value_twoside:.4f}")
print(f"p_value (right tail) = {p_value_right:.4f}")
print(f"p_value (left tail) = {p_value_left:.4f}")

degree of freedom =  54
t-value = 2.0824
p_value (two sided) = 0.0421
p_value (right tail) = 0.0210
p_value (left tail) = 0.9790


In our case, we are performing a right-tailed hypothesis test. The right-tailed p-value is 0.0210.
The p-value is smaller than the required significance level 0.05. As a result, we reject the null hypothesis in favor of the alternative hypothesis. 

We can conclude that that company 1's service results in higher tax payable.

<h2>Problem 13.130

令 Regular price 買 warranty 的機率為 p1。<br>
   Sale price 買 warranty 的機率為 p2。

Null Hypothesis: p1 - p2 = 0 <br>
Alternative Hypothesis: p1 - p2 > 0.

In [7]:
n1 = 229
n2 = 178

x1 = 47
x2 = 25

p1 = x1 / n1
p2 = x2 / n2

p_pool = (x1 + x2) / (n1 + n2)
sd_p_pool = (p_pool * (1 - p_pool) * (1 / n1 + 1 / n2)) ** 0.5
z_value = (p1 - p2) / sd_p_pool

print(f"p1 = {p1:.4f}")
print(f"p2 = {p2:.4f}")
print(f"sd_p_pool = {sd_p_pool:.4f}")
print(f"z-value = {z_value:.4f}")
p_value = 1 - stats.norm.cdf(z_value)
print(f"p-value = {p_value:.4f}")

p1 = 0.2052
p2 = 0.1404
sd_p_pool = 0.0381
z-value = 1.6992
p-value = 0.0446


In our case, we are performing a right-tailed hypothesis test. The p-value is 0.0446. The p-value is smaller than the required significance level 0.1. As a result, we reject the null hypothesis in favor of the alternative hypothesis. 

We can conclude that those who paid the regular price are more likely to buy an extended warranty.

<h2>Problem 13.156

In [8]:
# 讀取資料集
df1 = pd.read_excel('Xr13-156.xlsx')

# 簡單看一下資料集的樣子
# 如果這是你第一次使用這個資料集，使用Excel開啟資料集，
# 並確認程式讀到的跟你在Excel中看到的一樣
print('Head of dataset:')
display(df1.head())

print("Tail of dataset:")
display(df1.tail())

Head of dataset:


No religion  Religion  Wallonian
0            2         2          1
1            2         2          1
2            2         2          1
3            1         2          1
4            2         2          1

Tail of dataset:


No religion  Religion  Wallonian
3195            2         2          1
3196            2         2          1
3197            2         2          1
3198            2         2          1
3199            2         2          1

令 No religion call back 的機率為 p1。<br>
   Religion call back 的機率為 p2。

Null Hypothesis: p1 - p2 = 0 <br>
Alternative Hypothesis: p1 - p2 > 0.

In [9]:
r1 = df1["No religion"]
r2 = df1["Religion"]

r1 = r1 == 1
r2 = r2 == 1

In [10]:
p1 = r1.mean()
p2 = r2.mean()
r12 = np.concatenate([r1, r2])
p_pool = r12.mean()
sd_p_pool = (p_pool * (1 - p_pool) * 
             (1 / r1.shape[0] + 1 / r2.shape[0])) ** 0.5
z_value = (p1 - p2) / sd_p_pool
print(f"p1 = {p1:.4f}")
print(f"p2 = {p2:.4f}")
print(f"sd_p_pool = {sd_p_pool:.4f}")
print(f"z-value = {z_value:.4f}")
p_value = 1 - stats.norm.cdf(z_value)
print(f"p-value = {p_value:.4f}")

p1 = 0.0850
p2 = 0.0750
sd_p_pool = 0.0068
z-value = 1.4744
p-value = 0.0702


In our case, we are performing a right-tailed hypothesis test. The p-value is 0.0702. The p-value is larger than the required significance level 0.05. As a result, we reject the alternative hypothesis in favor of the null hypothesis. 

There is no sufficient evidence to infer that including a reference to religious activity reduces the probability of a call back.